Source/ARTHttp.m

```diff
@@ -175,18 +179,19 @@ - (void)executeRequest:(NSMutableURLRequest *)request completion:(void (^)(NSHTT
     [self.logger debug:@"%@ %@", request.HTTPMethod, request.URL.absoluteString];
     [self.logger verbose:@"Headers %@", request.allHTTPHeaderFields];
 
+    __weak typeof(self) weakSelf = self;
```

Comment:
That was something I noticed in a test that creates 50 clients and stores them in an array and when the `removeAll` (items) is called, those clients remained in the memory because they were waiting for the response.